packages/credential-store/src/types/filters.ts

```diff
@@ -0,0 +1,83 @@
+import { CredentialRole, DigitalCredential, DocumentType, FindDigitalCredentialArgs } from '@sphereon/ssi-sdk.data-store'
+
+/**
+ * Creates a filter to find a digital credential by its ID or hash.
+ *
+ * @param credentialRole - The role to filter by (e.g., ISSUER, HOLDER).
+ * @param idOrHash - The ID or hash of the credential to search for.
+ * @returns A FindDigitalCredentialArgs array for filtering by ID or hash.
+ */
+export const credentialIdOrHashFilter = (credentialRole: CredentialRole, idOrHash: string): FindDigitalCredentialArgs => [
+  {
+    id: idOrHash,
+    credentialRole,
+  },
+  {
+    hash: idOrHash,
+    credentialRole,
+  },
+]
+
+/**
+ * Creates a filter for verifiable credentials with a specific role.
+ *
+ * @param credentialRole - The role to filter by (e.g., ISSUER, HOLDER).
+ * @param withFilter - Optional additional filter criteria.
+ * @returns A FindDigitalCredentialArgs array for filtering verifiable credentials by role.
+ */
+export const verifiableCredentialForRoleFilter = (
```

Comment:
Why is this only available for VCs? We need this for C, VP, P as well, as all of these can have roles

Reply:
This is just a helper to avoid duplicate code and make it easier to fetch only VC's
If you need one of the other document types, don't use "verifiableCredentialForRoleFilter"
Create your own filter or create a verifiablePresentationForRoleFilter function when having duplicate code.